### PR TITLE
Add Fortran 2023 draft links

### PR DESCRIPTION
**Pull request by lstngr**
Having the F23 draft on the homepage would be nice!
Thanks!

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -13,6 +13,7 @@ Quick links:
     * Fortran 2003 standard: https://wg5-fortran.org/N1601-N1650/N1601.pdf
     * Fortran 2008 standard: https://j3-fortran.org/doc/year/10/10-007r1.pdf
     * Fortran 2018 standard: https://j3-fortran.org/doc/year/18/18-007r1.pdf
+    * Fortran 2023 standard: https://j3-fortran.org/doc/year/23/23-007r1.pdf
```
Comment (certik) [inline index.rst:16]: Shouldn't the link be: https://j3-fortran.org/doc/year/24/24-007.pdf

Comment (lstngr) [reply index.rst:16]: Of course, my bad, I went a bit too fast!

**Review by certik — APPROVED** [review]
Awesome, thanks!

**Comment by certik** [comment]
We need to currently manually update the page (https://github.com/certik/fortran90.org/issues/54). If you have time, you can send a PR. Otherwise I'll get to it later.

**Comment by lstngr** [comment]
I'm not so familiar with Github Actions, so I think I'll pass!

**Pull request merged**